File: 151-reverse-words-in-a-string/reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) {
        s = s.trim();
        String reverse[] = s.split("\\s+");
        reverseArray(reverse);
        return String.join(" ", reverse);
    }
    public static void reverseArray(String[] array) {
        int left = 0, right = array.length - 1;
        while (left < right) {
            String temp = array[left];
            array[left] = array[right];
            array[right] = temp;
            left++;
            right--;
        }
    }
}

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.ts
function reverseWords(s: string): string {
    let words = s.split(" "); //["one", "two", "three", "four"]
    
    words = words.filter(item => {
        return item !== '';
    });
    let end = words.length - 1;
    let start = 0;
    while (start < end) {
        let temp = words[start];
        words[start] = words[end];
        words[end] = temp;
        start++;
        end--;
    }
    
    return words.join(" ");
};